### PR TITLE
Netmiko plugin fix connection issue

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Netmiko plugin fix connection issue

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "results": "total 125096\ndrwxr-xr-x  9 vagrant vagrant      4096 Feb 26 20:32 .\ndrwxr-xr-x  3 root    root         4096 May 10  2020 ..\n-rw-------  1 vagrant vagrant     82391 Feb 26 23:09 .bash_history\n-rw-r--r--  1 vagrant vagrant       220 May 10  2020 .bash_logout\n-rw-r--r--  1 vagrant vagrant      3685 Sep 20 17:26 .bashrc\ndrwxr-xr-x  6 vagrant vagrant      4096 Sep 20 17:27 .cache\ndrwxr-xr-x  4 vagrant vagrant      4096 Sep 16 00:15 .gem\n-rw-r--r--  1 vagrant vagrant        24 Sep 20 17:22 .gitconfig\ndrwx------  3 vagrant vagrant      4096 Sep 16 15:16 .gnupg\ndrwxr-xr-x 10 root    root         4096 May  6  2019 go\n-rw-r--r--  1 root    root    127938445 Sep 16 00:13 go1.12.5.linux-amd64.tar.gz\ndrwxr-xr-x  4 vagrant vagrant      4096 Sep 20 17:27 go_work\ndrwx------  4 vagrant vagrant      4096 Nov 25 00:51 .local\n-rw-r--r--  1 vagrant vagrant       807 May 10  2020 .profile\ndrwx------  2 vagrant vagrant      4096 Nov  2 16:16 .ssh\n-rw-------  1 vagrant vagrant     13109 Feb 26 20:32 .viminfo"
}


```

<summary>
docker run --rm -i rapid7/netmiko:1.0.2 --debug run < tests/show_commands.json
</summary>
</details>
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "TitleValidator" failed!
        Cause: ("output key 'results' error.", ValidationException("Title contains a lowercase 'results' when it should not."))

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "show_commands": input "host", there is no example field.
In action "show_commands": input "command", there is no example field.
In action "configuration_commands": input "host", there is no example field.
In action "configuration_commands": input "command", there is no example field.
In connection: input "device_type", there is no example field.
In connection: input "credentials", there is no example field.
In connection: input "secret", there is no example field.
In connection: input "key", there is no example field.
In connection: input "host", there is no example field.
In connection: input "port", there is no example field.
In connection: input "verbose", there is no example field.


----
[*] Total time elapsed: 1117.7140000000002ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2021-02-26 23:16:22.971024

Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'secret'
   Severity: Low   Confidence: Medium
   Location: ./komand_netmiko/connection/schema.py:12
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
11          PORT = "port"
12          SECRET = "secret"
13          VERBOSE = "verbose"

--------------------------------------------------

Code scanned:
        Total lines of code: 436
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 1.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 1.0
                High: 0.0
Files skipped (0):
[FAIL] Fails bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check

```

<summary>
make validate
</summary>
</details>
<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
WARNING: Use of 'PluginException' or 'ConnectionTestException' is recommended when raising an exception.
violation: komand_netmiko/actions/configuration_commands/action.py: line,  29
violation: komand_netmiko/actions/show_commands/action.py: line,  29
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "TitleValidator" failed!
        Cause: ("output key 'results' error.", ValidationException("Title contains a lowercase 'results' when it should not."))

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "show_commands": input "host", there is no example field.
In action "show_commands": input "command", there is no example field.
In action "configuration_commands": input "host", there is no example field.
In action "configuration_commands": input "command", there is no example field.
In connection: input "device_type", there is no example field.
In connection: input "credentials", there is no example field.
In connection: input "secret", there is no example field.
In connection: input "key", there is no example field.
In connection: input "host", there is no example field.
In connection: input "port", there is no example field.
In connection: input "verbose", there is no example field.


----
[*] Total time elapsed: 6691.672ms

```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

**Review by mrinehart-r7 — DISMISSED** [review]

File: netmiko/help.md [hunk 1]
```diff
@@ -19,15 +19,20 @@ _This plugin does not contain any requirements._
...
+|device_type|string|None|True|Device type|['a10', 'accedian', 'alcatel_aos', 'alcatel_sros', 'arista_eos', 'aruba_os', 'avaya_ers', 'avaya_vsp', 'brocade_fastiron', 'brocade_netiron', 'brocade_nos', 'brocade_vdx', 'brocade_vyos', 'checkpoint_gaia', 'calix_b6', 'ciena_saos', 'cisco_asa', 'cisco_ios', 'cisco_nxos', 'cisco_s300', 'cisco_tp', 'cisco_wlc', 'cisco_xe', 'cisco_xr', 'coriant', 'dell_force10', 'dell_powerconnect', 'eltex', 'enterasys', 'extreme', 'extreme_wing', 'f5_ltm', 'fortinet', 'generic_termserver', 'hp_comware', 'hp_procurve', 'huawei', 'juniper', 'juniper_junos', 'linux', 'mellanox', 'mrv_optiswitch', 'netapp_cdot', 'ovs_linux', 'paloalto_panos', 'pluribus', 'quanta_mesh', 'ruckus_fastiron', 'ubiquiti_edge', 'ubiquiti_edgeswitch', 'vyatta_vyos', 'vyos']|None|
+|host|string|None|True|Remote Host|None|None|
+|key|credential_asymmetric_key|None|False|A base64 encoded SSH private key to use to authenticate to network device|None|None|
+|port|integer|22|True|Remote port|None|None|
+|secret|credential_secret_key|None|False|Secret password|None|None|
+|verbose|boolean|False|True|Additional messages to standard output|None|None|
+
+Example input:
```
Comment (jmcadams-r7) [inline netmiko/help.md:32]: Need an example input.

**Review by jmcadams-r7 — CHANGES_REQUESTED** [review]

File: netmiko/help.md (hunk 1 above)
Comment (r7-kszczepanskagorna) [reply netmiko/help.md:32]: Hi @jmcadams-r7 :) We can create a ticket for  plugin audit and then for improvments which also will include example inputs. Is it ok for you? :)

Comment (jschipp-r7) [reply netmiko/help.md:32]: Yes, that's fine @r7-kszczepanskagorna

File: netmiko/help.md [hunk 2]
```diff
@@ -101,6 +116,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 1.0.2 - Fix connection
```
Comment (jschipp-r7) [inline netmiko/help.md:119]:
```suggestion
* 1.0.2 - Fix bug where the SSH private key was not being retrieved correctly from the user in the connection
```

**Review by jschipp-r7 — APPROVED** [review]
